### ISSUE TITLE
LRUCache.set not update cache value when the key exists

### ISSUE THREAD

**Issue by CwjXFH** [msiemens/tinydb#560]
[LRUCache.set](https://github.com/msiemens/tinydb/blob/master/tinydb/utils.py#L102) does not update the cache value when the key exists, this may confuse the user. Why is it designed this way?
```py
def set(self, key: K, value: V):
    if self.cache.get(key):
        self.cache.move_to_end(key, last=True)
    else:
        self.cache[key] = value
        if self.capacity is not None and self.length > self.capacity:
            self.cache.popitem(last=False)
```

**Comment by msiemens** [comment]
Good point! Would you like to contribute a bugfix on this?

**Comment by CwjXFH** [comment]
fix: https://github.com/msiemens/tinydb/pull/573

**Comment by msiemens** [comment]
Fixed in #573

**Comment by msiemens** [comment]
v4.8.2 includes this fix and has been released
